### PR TITLE
[SPARK-18535][SPARK-19720][CORE][BACKPORT-2.1] Redact sensitive information

### PR DESCRIPTION
**Pull request by dmvieira**
## What changes were proposed in this pull request?

Backporting SPARK-18535 and SPARK-19720 to spark 2.1

It's a backport PR that redacts senstive information by configuration to Spark UI and Spark Submit console logs.

Using reference from Mark Grover mark@apache.org PRs

## How was this patch tested?

Same tests from PR applied

### CODE REVIEW

**Comment by vanzin** [comment]
ok to test

**Comment by dmvieira** [comment]
ok to test

**Comment by dmvieira** [comment]
I don't know why these tests are breaking. Could some one help me?

Permission denied?

**Comment by gatorsmile** [comment]
cc @markgrover @vanzin Could you please take a look at this?

File: dev/run-tests.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def determine_modules_to_test(changed_modules):
     if modules.root in modules_to_test:
         return [modules.root]
     return toposort_flatten(
-        {m: set(m.dependencies).intersection(modules_to_test) for m in modules_to_test}, sort=True)
+        dict((m, set(m.dependencies).intersection(modules_to_test)) for m in modules_to_test))
```
Comment (vanzin) [inline dev/run-tests.py]: I see you added these changes to fix the tests, but they're unrelated to the patches you're backporting.

They should, at the very least, be a separate PR, if they're really needed.

Comment (dmvieira) [inline dev/run-tests.py]: I can remove it, but tests will fail at Jenkins

Comment (vanzin) [inline dev/run-tests.py]: Then, as I suggested, open a separate PR with the fix for the tests.

**Comment by dmvieira** [comment]
I removed test fixes and add another PR: https://github.com/apache/spark/pull/18873 from this branch

**Comment by vanzin** [comment]
Merging to 2.1.

**Comment by vanzin** [comment]
@dmvieira please close the PR since github doesn't do it automatically.

**Comment by dmvieira** [comment]
Thank you @vanzin

**Comment by markgrover** [comment]
Thanks @dmvieira and @vanzin
